r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -711,3 +711,48 @@ test_that("am/pm mirror lubridate", {
   )
 
 })
+
+test_that("semester works with temporal types", {
+  test_df <- tibble(
+    month_as_int = c(1:12, NA),
+    month_as_char_pad = ifelse(month_as_int < 10, paste0("0", month_as_int), month_as_int),
+    dates = as.Date(paste0("2021-", month_as_char_pad, "-15"))
+  )
```

Comment:
I see this repeated down below as well and we don't use all of the inputs here. We should either curate the inputs so that this tibble has all and only the columns that are used in this block (and same with the one below) or we should factor these out + put them up top or somewhere else such that we define it once and then use it in multiple `test_that` blocks.